compaction/compaction.cc

```diff
@@ -1473,7 +1473,7 @@ class scrub_compaction final : public regular_compaction {
 
     reader_consumer make_interposer_consumer(reader_consumer end_consumer) override {
         return [this, end_consumer = std::move(end_consumer)] (flat_mutation_reader reader) mutable -> future<> {
-            return mutation_writer::segregate_by_partition(std::move(reader), std::move(end_consumer));
+            return mutation_writer::segregate_by_partition(std::move(reader), 100, std::move(end_consumer));
```

Comment:
100 open buckets still seems very high
why not makes it much smaller? like 2 or 10?

Reply:
The only reason is that this is the existing limit for the TWCS's timestamp based splitter. That number is also just a magic number without any strong arguments to support it.
I can change it to be lower but I think if we deem 100 as too much, we should also change the timestamp based splitter's limit as well.